src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -525,6 +585,418 @@ public List<UserType> fetchUserTypes() {
+        }
+
+        if (isTrue(caseWorkerProfile.getSuspended())) {
+            //when existing profile with delete flag is true then log exception add entry in exception table
+            staffProfileAuditService.saveStaffAudit(AuditStatus.FAILURE, ALREADY_SUSPENDED_ERROR_MESSAGE,
+                    caseWorkerProfile.getCaseWorkerId(), profileRequest, STAFF_PROFILE_UPDATE);
+            throw new StaffReferenceException(HttpStatus.BAD_REQUEST, StringUtils.EMPTY,
+                    ALREADY_SUSPENDED_ERROR_MESSAGE);
+        }
+        return caseWorkerProfile;
+
+    }
+
+    public UserProfileResponse getUserProfileFromUP(String idamId) {
+        Response response = userProfileFeignClient.getUserProfileWithRolesById(idamId);
+        ResponseEntity<Object> responseEntity = toResponseEntity(response, UserProfileResponse.class);
+
+        Optional<Object> resultResponse = validateAndGetResponseEntity(responseEntity);
+        if (resultResponse.isPresent() && resultResponse.get() instanceof UserProfileResponse profileResponse) {
+            return profileResponse;
+        }
+        return null;
+    }
+
+    private CaseWorkerProfile checkStaffProfileForUpdate(StaffProfileCreationRequest profileRequest) {
+
+        // get all existing profile from db (used IN clause)
+        CaseWorkerProfile caseWorkerProfile = caseWorkerProfileRepo
+                .findByEmailId(profileRequest.getEmailId().toLowerCase());
+        if (caseWorkerProfile == null) {
+            staffProfileAuditService.saveStaffAudit(AuditStatus.FAILURE,PROFILE_NOT_PRESENT_IN_DB,
+                    StringUtils.EMPTY,profileRequest,STAFF_PROFILE_UPDATE);
+            throw new StaffReferenceException(HttpStatus.BAD_REQUEST,StringUtils.EMPTY,
+                    PROFILE_NOT_PRESENT_IN_DB);
+        }
+
+
+        return caseWorkerProfile;
+
+    }
+
+
+    public CaseWorkerProfile updateStaffProfiles(StaffProfileCreationRequest cwUiRequest,
+                                                       CaseWorkerProfile caseWorkerProfile) {
+
+        CaseWorkerProfile processedCwProfile;
+
+        try {
+
+            //process update and suspend CW profiles
+            CaseWorkerProfile cwProfileToPersist = null;
+            cwProfileToPersist = processExistingCaseWorkers(
+                    cwUiRequest, caseWorkerProfile);
+            // persist in db
+
+
+            processedCwProfile = persistCaseWorker(cwProfileToPersist,
+                    cwUiRequest);
+        } catch (Exception exp) {
+            log.error("{}:: createCaseWorkerUserProfiles failed  ::{}", loggingComponentName,
+                    exp);
+            throw exp;
+        }
+        return processedCwProfile;
+    }
+
+    public CaseWorkerProfile processExistingCaseWorkers(
+            StaffProfileCreationRequest cwUiRequest, CaseWorkerProfile caseWorkerProfiles) {
+
+        CaseWorkerProfile filteredProfile = null;
+
+        if (isTrue(caseWorkerProfiles.getSuspended())) {
```

Comment:
You have already validated this in https://github.com/hmcts/rd-caseworker-ref-api/pull/609/files#diff-1e81f87fca954d43c994f1f944b28ade4e446be6a0c98654292374200f92066aR636